beacon_node/lighthouse_network/src/discovery/enr_ext.rs

```diff
@@ -196,45 +198,42 @@ impl CombinedKeyPublicExt for CombinedPublicKey {
     ///
     /// This is only available with the `libp2p` feature flag.
     fn as_peer_id(&self) -> PeerId {
+        use libp2p::identity::{ed25519, secp256k1};
         match self {
             Self::Secp256k1(pk) => {
                 let pk_bytes = pk.to_sec1_bytes();
-                let libp2p_pk = libp2p::core::PublicKey::Secp256k1(
-                    libp2p::core::identity::secp256k1::PublicKey::decode(&pk_bytes)
-                        .expect("valid public key"),
-                );
+                let libp2p_pk: PublicKey = secp256k1::PublicKey::try_from_bytes(&pk_bytes)
+                    .expect("valid public key")
+                    .into();
                 PeerId::from_public_key(&libp2p_pk)
             }
             Self::Ed25519(pk) => {
                 let pk_bytes = pk.to_bytes();
-                let libp2p_pk = libp2p::core::PublicKey::Ed25519(
-                    libp2p::core::identity::ed25519::PublicKey::decode(&pk_bytes)
-                        .expect("valid public key"),
-                );
+                let libp2p_pk: PublicKey = ed25519::PublicKey::try_from_bytes(&pk_bytes)
+                    .expect("valid public key")
+                    .into();
                 PeerId::from_public_key(&libp2p_pk)
             }
         }
     }
 }
 
 impl CombinedKeyExt for CombinedKey {
-    fn from_libp2p(key: &libp2p::core::identity::Keypair) -> Result<CombinedKey, &'static str> {
-        match key {
-            Keypair::Secp256k1(key) => {
-                let secret =
-                    discv5::enr::k256::ecdsa::SigningKey::from_slice(&key.secret().to_bytes())
-                        .expect("libp2p key must be valid");
-                Ok(CombinedKey::Secp256k1(secret))
-            }
-            Keypair::Ed25519(key) => {
-                let ed_keypair = discv5::enr::ed25519_dalek::SigningKey::from_bytes(
-                    &(key.encode()[..32])
-                        .try_into()
-                        .expect("libp2p key must be valid"),
-                );
-                Ok(CombinedKey::from(ed_keypair))
-            }
-            Keypair::Ecdsa(_) => Err("Ecdsa keypairs not supported"),
+    fn from_libp2p(key: Keypair) -> Result<CombinedKey, &'static str> {
+        // TODO: more clones that should not be happening.
```

Comment:
As you probably noticed, https://github.com/libp2p/rust-libp2p/issues/3649 is now resolved. Let me know if you'd like a patch release for that.